core/src/main/kotlin/org/evomaster/core/problem/rest/service/RestResourceFitness.kt

```diff
@@ -58,43 +71,97 @@ class RestResourceFitness : AbstractRestFitness<RestIndividual>() {
         //run the test, one action at a time
         var indexOfAction = 0
 
+        RCallsLoop@
         for (call in individual.getResourceCalls()) {
 
-            val result = doDbCalls(call.seeActions(ActionFilter.ONLY_SQL) as List<DbAction>, sqlIdMap, failureBefore, executedDbActions, actionResults)
+            val result = doDbCalls(
+                call.seeActions(ActionFilter.ONLY_SQL) as List<DbAction>,
+                sqlIdMap,
+                failureBefore,
+                executedDbActions,
+                actionResults
+            )
             failureBefore = failureBefore || result
 
             var terminated = false
 
-            for (a in call.seeActions(ActionFilter.NO_SQL)){
+            for(a in call.getViewOfChildren().filterIsInstance<EnterpriseActionGroup>()){
+                // Note: [indexOfAction] is used to register the action in RemoteController
+                //  to map it to the ActionDto.
+
+                /*
+                    Always need to reset WireMock.
+                    Assume there no is External Action here. Still, the SUT could make a call to an external service
+                    because new code is reached (eg due mutation of some genes).
+                    We do not want such new call to re-use a mocking from a previous action which were left on.
+                    So 2 options: (1) always reset before calling SUT, or (2) always reset after call in which external
+                    setups were used.
+                 */
+                externalServiceHandler.resetServedRequests()
+
+                val externalServiceActions = a.getExternalServiceActions()
+
+                externalServiceActions.filterIsInstance<HttpExternalServiceAction>().forEach {
```

Comment:
1) before building the response, might `resetActive` first, i.e., set `active` based on `used`
2) when building the response, filter the actions whose `active` is true

Reply:
2. is already there, added `resetActive`

Reply:
1 should before this `forEach`, otherwise it is too late.
```
externalServiceActions.forEach{it.resetActive()}
externalServiceActions.filter{it.active}.filterIsInstance<HttpExternalServiceAction>().forEach{...}
```

Reply:
What about when we create the action?